internal/api/mfa.go

```diff
@@ -174,6 +258,85 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	})
 }
 
+func (a *API) challengePhoneFactor(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	config := a.config
+	db := a.db.WithContext(ctx)
+	user := getUser(ctx)
+	factor := getFactor(ctx)
+	ipAddress := utilities.GetIPAddress(r)
+	params := &ChallengeFactorParams{}
+	if err := retrieveRequestParams(r, params); err != nil {
+		return err
+	}
+	channel := params.Channel
+
+	if channel == "" {
+		channel = sms_provider.SMSProvider
+	}
+	smsProvider, err := sms_provider.GetSmsProvider(*config)
+	if err != nil {
+		return internalServerError("Failed to get SMS provider").WithInternalError(err)
+	}
+	if !sms_provider.IsValidMessageChannel(channel, config.Sms.Provider) {
+		return badRequestError(ErrorCodeValidationFailed, InvalidChannelError)
+	}
+
+	otp, err := crypto.GenerateOtp(config.MFA.Phone.OtpLength)
+	if err != nil {
+		panic(err)
+	}
+	challenge, err := factor.CreateSMSChallenge(ipAddress, otp, config.Security.DBEncryption.Encrypt, config.Security.DBEncryption.EncryptionKeyID, config.Security.DBEncryption.EncryptionKey)
+	if err != nil {
+		return internalServerError("error creating SMS Challenge")
+	}
+	message, err := generateSMSFromTemplate(config.MFA.Phone.SMSTemplate, otp)
+	if err != nil {
+		return internalServerError("error generating sms template").WithInternalError(err)
+	}
+	if config.Hook.SendSMS.Enabled {
+		input := hooks.SendSMSInput{
+			User: user,
+			SMS: hooks.SMS{
+				OTP:     otp,
+				SMSType: "mfa",
+			},
+		}
+		output := hooks.SendSMSOutput{}
+		err := a.invokeHook(a.db, r, &input, &output, a.config.Hook.SendSMS.URI)
+		if err != nil {
+			return internalServerError("error invoking hook")
+		}
+	} else {
+		if !challenge.SentAt.Add(config.MFA.Phone.MaxFrequency).Before(time.Now()) {
+			return MaxFrequencyLimitError
+		}
+
+		messageID, err := smsProvider.SendMessage(string(factor.Phone), message, channel, otp)
+		if err != nil {
+			return internalServerError(fmt.Sprintf("error sending message: %v", messageID)).WithInternalError(err)
```

Comment:
Is this really something that needs to be returned to the user like "error sending message: 9238420984393"?